From 10c234bc645fde4aea7a45ebc5e68b2cf4d42197 Mon Sep 17 00:00:00 2001
From: Irina Tirdea <irina.tirdea@nxp.com>
Date: Thu, 2 Feb 2017 15:35:48 +0200
Subject: [PATCH 1391/5242] MLK-13793-2 regulator: anatop: Do not print error
 message for EPROBE_DEFER

commit  05d59c198b0e23a89e06d3e3b7fc6f811c547270 from
https://source.codeaurora.org/external/imx/linux-imx.git

An anatop regulator will return -EPROBE_DEFER when trying to register,
if it has a supply from PMIC and this supply is not yet registered.
This does not represent an error since the driver will call probe again later,
so skip the error message in this case.

Signed-off-by: Irina Tirdea <irina.tirdea@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/regulator/anatop-regulator.c |    7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/regulator/anatop-regulator.c b/drivers/regulator/anatop-regulator.c
index 23dfcd9..2f6754b 100644
--- a/drivers/regulator/anatop-regulator.c
+++ b/drivers/regulator/anatop-regulator.c
@@ -370,9 +370,10 @@ static int anatop_regulator_probe(struct platform_device *pdev)
 	/* register regulator */
 	rdev = devm_regulator_register(dev, rdesc, &config);
 	if (IS_ERR(rdev)) {
-		dev_err(dev, "failed to register %s\n",
-			rdesc->name);
-		return PTR_ERR(rdev);
+		ret = PTR_ERR(rdev);
+		if (ret != -EPROBE_DEFER)
+			dev_err(dev, "failed to register %s\n",	rdesc->name);
+		return ret;
 	}
 
 	platform_set_drvdata(pdev, rdev);
-- 
1.7.9.5

